config.go

```diff
@@ -29,42 +29,62 @@ import (
 	yaml "gopkg.in/yaml.v2"
 )
 
+const deviceConfigPath = "/etc/cacophony/device.yml"
+
 type Config struct {
-	ServerURL  string `yaml:"server-url"`
-	Group      string `yaml:"group"`
-	DeviceName string `yaml:"device-name"`
-	DeviceID   int    `yaml:"device-id"`
+	ServerURL  string `yaml:"server-url" json:"serverURL"`
+	Group      string `yaml:"group" json:"groupname"`
+	DeviceName string `yaml:"device-name" json:"devicename"`
+	DeviceID   int    `yaml:"device-id" json:"deviceID"`
 }
 
 type PrivateConfig struct {
 	Password string `yaml:"password"`
 }
 
+// CreateDeviceConfig from filename, only if overwrite set or doesn't exist
+func CreateDeviceConfig(filename string, overwrite bool) (*Config, error) {
+	_, err := os.Stat(deviceConfigPath)
+	conf := &Config{}
+	if err != nil || overwrite {
+		conf, err = ParseConfigFile(filename)
+		if err != nil {
+			return nil, err
+		}
+		err = conf.Save()
+	} else {
+		conf, err = LoadConfig()
+	}
+	return conf, err
+}
+
 //Validate checks supplied Config contains the required data
 func (conf *Config) Validate() error {
 	if conf.ServerURL == "" {
 		return errors.New("server-url missing")
 	}
-	if conf.Group == "" {
-		return errors.New("group missing")
-	}
-	if conf.DeviceName == "" {
-		return errors.New("device-name missing")
+
+	if (conf.Group == "" || conf.DeviceName == "") && conf.DeviceID == 0 {
+		return errors.New("device-info missing")
 	}
 	return nil
 }
 
-//SaveToFile writes config to supplied filename
-func (conf *Config) SaveToFile(filename string) error {
+//Save writes config to deviceConfig
+func (conf *Config) Save() error {
 	buf, err := yaml.Marshal(&conf)
 	if err != nil {
 		return err
 	}
+	return ioutil.WriteFile(deviceConfigPath, buf, 0600)
```

Comment:
There's a small chance that multiple API client might try to write to device.yaml at the same time. There should some file locking as there is for password file.